src/Dofs/ConstraintHandler.jl

```diff
@@ -426,18 +486,193 @@ function apply!(KK::Union{SparseMatrixCSC,Symmetric}, f::AbstractVector, ch::Con
+                    #end
+                end
+            end
+        else    
+            for a in nzrange(K, col)
+                row = K.rowval[a]
+                if distribute[row] == 0
+                    lc = lcs[distribute[col]]
+                    for (d,v) in lc.entries
+                        Kval = K.nzval[a]
+                        _addindex_sparsematrix!(K, v*Kval, row, d)
+                    end
+                else
+                    lc1 = lcs[distribute[col]]
+                    for (d1,v1) in lc1.entries
+                        lc2 = lcs[distribute[row]]
+                        for (d2,v2) in lc2.entries
+                            Kval = K.nzval[a]
+                            _addindex_sparsematrix!(K, v1*v2*Kval, d1, d2)
+                        end
+                    end
+                end
+            end
+
+            if condense_f
+                lc = lcs[distribute[col]]
+                for (d,v) in lc.entries
+                    f[d] += f[col] * v
+                end
+                f[lc.constrained_dof] = 0.0
+            end
+        end
+    end
+end
+
+# Copied from SparseArrays._setindex_scalar!(...)
+# Custom SparseArrays._setindex_scalar!() that throws error if entry K(_i,_j) does not exist
+function _addindex_sparsematrix!(A::SparseMatrixCSC{Tv,Ti}, v::Tv, i::Ti, j::Ti) where {Tv, Ti}
+    if !((1 <= i <= size(A, 1)) & (1 <= j <= size(A, 2)))
+        throw(BoundsError(A, (i,j)))
+    end
+    coljfirstk = Int(SparseArrays.getcolptr(A)[j])
+    coljlastk = Int(SparseArrays.getcolptr(A)[j+1] - 1)
+    searchk = searchsortedfirst(rowvals(A), i, coljfirstk, coljlastk, Base.Order.Forward)
+    if searchk <= coljlastk && rowvals(A)[searchk] == i
+        # Column j contains entry A[i,j]. Update and return
+        nonzeros(A)[searchk] += v
+        return A
+    end
+    error("Sparsity pattern missing entries for the condensation pattern. Make sure to call `create_sparsity_pattern(dh::DofHandler, ch::ConstraintHandler) when using linear constraints.`")
+end
+
+#A[i,j] += 0.0 does not add entries to sparse matrices, so we need to first add 1.0, and then remove it
+function add_entry!(A::SparseMatrixCSC, i::Int, j::Int)
+    if A[i,j] == 0.0 # Check first if zero to not remove already non-zero entries
+        A[i,j] = 1.0
+        A[i,j] = 0.0
+    end
+end
+
+"""
+    create_constraint_matrix(ch::ConstraintHandler)
+
+Creates and returns the constraint matrix, C, and the inhomogeneities, g, from the linear and Dirichlet constraints.
+
+The constraint matrix relates constrained and free degrees of freedom via a = C*a_f + g. The 
+condensed system of linear equations is obtained as C'*K*C and C'*(f - K*g)
+"""
+function create_constraint_matrix(ch::ConstraintHandler{dh,T}) where {dh,T}
+    @assert(isclosed(ch))
+
+    distribute = zeros(Int, ndofs(ch.dh))
+    for (i,d) in enumerate(ch.free_dofs)
+        distribute[d] = i;
+    end
+
+    I = Int[]; J = Int[]; V = Int[];
```

Comment:
```suggestion
    I = Int[]; J = Int[]; V = T[];
```
(This only worked because you have integer valued factors in your tests)